File: backend/src/models/Persona.js
const { pool } = require('../config/db');

class Persona {
  /**
   * Crear una nueva persona
   */
  static async create({ nombres, apellido_paterno, apellido_materno = null, ci, ci_expedido = 'LP', sexo = 'M', estado_civil = null, telefono = null, email = null, empresa_telefonica = null, direccion = null }) {
    const [result] = await pool.query(`
      INSERT INTO personas (nombres, apellido_paterno, apellido_materno, ci, ci_expedido, sexo, estado_civil, telefono, email, empresa_telefonica, direccion, activo)
      VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, 1)
    `, [nombres, apellido_paterno, apellido_materno, ci, ci_expedido, sexo, estado_civil, telefono, email, empresa_telefonica, direccion]);

    return this.findById(result.insertId);
  }

  /**
   * Buscar persona por ID
   */
  static async findById(id) {
    const [rows] = await pool.query('SELECT * FROM personas WHERE id = ?', [id]);
    return rows[0] || null;
  }

  /**
   * Buscar persona por Cédula de Identidad (CI)
   */
  static async findByCi(ci) {
    const [rows] = await pool.query('SELECT * FROM personas WHERE ci = ?', [ci]);
    return rows[0] || null;
  }

  /**
   * Listar todas las personas activas
   */
  static async findAll({ search = '', activo = true, limit = 50, offset = 0 } = {}) {
    let sql = 'SELECT * FROM personas WHERE 1=1';
    const params = [];

    if (activo !== null) {
      sql += ' AND activo = ?';
      params.push(activo ? 1 : 0);
    }

    if (search) {
      sql += ' AND (nombres LIKE ? OR apellido_paterno LIKE ? OR apellido_materno LIKE ? OR ci LIKE ?)';
      const term = `%${search}%`;
      params.push(term, term, term, term);
    }

    sql += ' ORDER BY apellido_paterno ASC, nombres ASC LIMIT ? OFFSET ?';
    params.push(parseInt(limit, 10), parseInt(offset, 10));

    const [rows] = await pool.query(sql, params);
    return rows;
  }

  /**
   * Actualizar datos de persona
   */
  static async update(id, data) {
    const fields = [];
    const params = [];

    const allowed = ['nombres', 'apellido_paterno', 'apellido_materno', 'ci', 'ci_expedido', 'sexo', 'estado_civil', 'telefono', 'email', 'empresa_telefonica', 'direccion', 'activo'];
    for (const key of allowed) {
      if (data[key] !== undefined) {
        fields.push(`\`${key}\` = ?`);
        params.push(data[key]);
      }
    }

    if (fields.length === 0) return this.findById(id);

    params.push(id);
    await pool.query(`UPDATE personas SET ${fields.join(', ')} WHERE id = ?`, params);
    return this.findById(id);
  }

  /**
   * Borrado lógico con verificación de dependencias en usuarios
   */
  static async softDelete(id) {
    const [usuarios] = await pool.query('SELECT id FROM usuarios WHERE persona_id = ? AND activo = 1', [id]);
    if (usuarios.length > 0) {
      throw new Error('No se puede eliminar la persona: tiene un usuario activo asociado en el sistema.');
    }

    await pool.query('UPDATE personas SET activo = 0 WHERE id = ?', [id]);
    return true;
  }
}

module.exports = Persona;

File: backend-dotnet/GestionDocumental.Api/Entities/Persona.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace GestionDocumental.Api.Entities
{
    [Table("personas")]
    public class Persona
    {
        [Key]
        [Column("id")]
        public int Id { get; set; }

        [Required]
        [MaxLength(100)]
        [Column("nombres")]
        public string Nombres { get; set; } = string.Empty;

        [Required]
        [MaxLength(100)]
        [Column("apellido_paterno")]
        public string ApellidoPaterno { get; set; } = string.Empty;

        [MaxLength(100)]
        [Column("apellido_materno")]
        public string? ApellidoMaterno { get; set; }

        [Required]
        [MaxLength(20)]
        [Column("ci")]
        public string Ci { get; set; } = string.Empty;

        [Required]
        [MaxLength(5)]
        [Column("ci_expedido")]
        public string CiExpedido { get; set; } = "CH";

        [Required]
        [MaxLength(10)]
        [Column("sexo")]
        public string Sexo { get; set; } = "M";

        [MaxLength(20)]
        [Column("estado_civil")]
        public string? EstadoCivil { get; set; }

        [MaxLength(20)]
        [Column("telefono")]
        public string? Telefono { get; set; }

        [MaxLength(150)]
        [Column("email")]
        public string? Email { get; set; }

        [MaxLength(20)]
        [Column("empresa_telefonica")]
        public string? EmpresaTelefonica { get; set; }

        [MaxLength(255)]
        [Column("direccion")]
        public string? Direccion { get; set; }

        [Column("activo")]
        public bool Activo { get; set; } = true;

        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Column("updated_at")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Relaciones
        public virtual ICollection<Usuario> Usuarios { get; set; } = new List<Usuario>();
    }
}
